main.go

```diff
@@ -15,19 +15,24 @@ import (
 func Digest(ctx context.Context, e firestore.DocumentEventData) error {
 	log.Printf("→ firestore protobuf document: %v", e)
 
-	b, err := firestoreconv.Report(e.Value)
+	r, err := firestoreconv.Report(e.Value)
 	if err != nil {
 		return err
 	}
 
-	data := b.Benchmark.Times()
+	data := r.Benchmark.Times()
 
 	s, err := stats.Compute(data)
 	if err != nil {
 		return err
 	}
 
-	log.Printf("→ computed stats: %v", s)
+	b, err := s.MarshalJSON()
+	if err != nil {
+		log.Printf("warning: failed to marshal stats: %s", err)
+		return nil
+	}
```

Comment:
[question] Why print and return `nil` for that error specifically instead of `return fmt.Errorf(...)`?

Reply:
Because it's not a _fail_ of the cloud function, just the log. If we keep this i will add a comment.

Reply:
Well, not a _direct fail_ of the cloud function, but if a marshaling error happens here it's probably that something very weird happened to that stats struct before (I think it just can't happen actually).
Anyway it's not on the table anymore with the new (better!) implem 🙂